dashboard/pages/5_Word_Clouds.py

```diff
@@ -135,7 +149,29 @@ def run_app():
 
     st.title("Article Content Word Cloud by News Source")
 
-    fn_articles, dn_articles = get_all_article_content()
+    time_range_options = ["Last hour", "Last 24 hours", "Last 7 days"]
+    selected_time_range = st.sidebar.select_slider(
+        "Select time range",
+        options=time_range_options,
+        value="Last 24 hours"
+    )
+
+    time_range_mapping = {
+        "Last hour": 0,
+        "Last 24 hours": 1,
+        "Last 7 days": 2
+    }
+    time_range = time_range_mapping[selected_time_range]
+
+    st.write(f"Time Range Selected: {selected_time_range}")
+
+    articles = get_all_article_content()
+    filtered_articles = filter_articles_by_date(articles, time_range)
+
+    fn_articles = [
+        article["article_content"] for article in filtered_articles if article["source_name"] == "Fox News"]
+    dn_articles = [
+        article["article_content"] for article in filtered_articles if article["source_name"] == "Democracy Now!"]
```

Comment:
DRY refactoring could be used here since both the lines use the same logic

Reply:
Will do this, thanks!